src/System.Collections/src/System/Collections/BitArray.cs

```diff
@@ -44,13 +44,16 @@ public BitArray(int length, bool defaultValue)
                 throw new ArgumentOutOfRangeException(nameof(length), length, SR.ArgumentOutOfRange_NeedNonNegNum);
             }
 
-            m_array = new int[GetArrayLength(length, BitsPerInt32)];
+            m_array = new int[GetIntegerArrayLengthFromBitLength(length)];
             m_length = length;
 
-            int fillValue = defaultValue ? unchecked(((int)0xffffffff)) : 0;
-            for (int i = 0; i < m_array.Length; i++)
+            if (defaultValue)
             {
-                m_array[i] = fillValue;
+                int fillValue = unchecked((int)0xffffffff);
+                for (int i = 0; i < m_array.Length; i++)
+                {
+                    m_array[i] = fillValue;
+                }
```

Comment:
Could/should we use Span.Fill here instead? e.g.
```C#
if (defaultValue)
{
    new Span<int>(m_array).Fill(unchecked((int)0xffffffff);
}
```
I'm not sure what the break even point would be.

Reply:
Yep, tried. Not great for small inputs.

Reply:
I tried the following:
```C#
        private static void Fill1(int[] m_array)
        {
            int fillValue = unchecked(((int)0xffffffff));
            for (int i = 0; i < m_array.Length; i++)
            {
                m_array[i] = fillValue;
            }
        }

        private static void Fill2(int[] m_array)
        {
            Span<sbyte> byteSpan = MemoryMarshal.Cast<int, sbyte>(m_array.AsSpan());
            byteSpan.Fill(-1);
        }

        private static void Fill3(int[] m_array)
        {
            int fillValue = unchecked(((int)0xffffffff));
            m_array.AsSpan().Fill(fillValue);
        }
```